src/Workspaces/Core/Portable/Workspace/Solution/SolutionCompilationState.CompilationTracker.cs

```diff
@@ -647,32 +653,6 @@ private Compilation CreateEmptyCompilation()
                 }
             }
 
-            /// <summary>
-            /// Attempts to get (without waiting) a metadata reference to a possibly in progress
-            /// compilation. Only actual compilation references are returned. Could potentially 
-            /// return null if nothing can be provided.
-            /// </summary>
-            public MetadataReference? GetPartialMetadataReference(ProjectState fromProject, ProjectReference projectReference)
-            {
-                if (ProjectState.LanguageServices == fromProject.LanguageServices)
-                {
-                    // if we have a compilation and its the correct language, use a simple compilation reference in any
-                    // state it happens to be in right now
-                    if (ReadState() is CompilationTrackerState compilationState)
```

Comment:
this was the crux of the bug.  the CompilationTrackerState we fetched here would have the compilations containing the source-documents, but not the referenced metadata (either PE references, or P2P references).  

I considered updating this logic to add that in.  But that would then effectively be doing what FinalizeCompilation already does.  So i uniformly handled it at that layer.